src/StudyApi/DependencyRegistrations/InfrastructureRegistration.cs

```diff
@@ -81,10 +84,15 @@ public static IServiceCollection AddInfrastructure(this IServiceCollection servi
             services.AddHttpClientWithRetry<IReferenceDataApiClient, ReferenceDataApiClient>(clientsSettings.ReferenceDataService, 2, logger);
             services.AddHttpClientWithRetry<IParticipantApiClient, ParticipantApiClient>(clientsSettings.ParticipantService, 2, logger);
 
+            // get DevSwitches from appsettings.json
+            var devSwitches = configuration.GetSection(DevSwitches.SectionName).Get<DevSwitches>();
+
             // If not Prod, then enable stubs
-            if (!ProdEnvironmentNames.Any(x => string.Equals(x, environmentName, StringComparison.OrdinalIgnoreCase)))
+            if (devSwitches.EnableStubs)
```

Comment:
I would recommend having an additional environment check to ensure this doesn't leak into production.

`if (devSwitches.EnableStubs && !ProdEnvironmentNames.Any(x => string.Equals(x, environmentName, StringComparison.OrdinalIgnoreCase)))`